client/my-sites/plans/jetpack-plans/upsell/index.tsx

```diff
@@ -0,0 +1,176 @@
+	);
+	const upsellCheckoutURL = useMemo(
+		() => ( upsellSlug ? buildCheckoutURL( siteSlug, upsellSlug, urlQueryArgs ) : '' ),
+		[ siteSlug, upsellSlug, urlQueryArgs ]
+	);
+	const isUpsellOwned = useMemo(
+		() =>
+			isFetchingPurchases ? false : !! purchases?.find( ( p ) => p.productSlug === upsellSlug ),
+		[ upsellSlug, purchases, isFetchingPurchases ]
+	);
+	const { intro, description, cta, features } = useMemo( () => {
+		if ( PLAN_JETPACK_SECURITY_T1_YEARLY === upsellSlug ) {
+			return {
+				intro: translate( 'Here’s a popular bundle for comprehensive site security:' ),
+				description: translate( 'Save money with all Jetpack security products in one bundle:' ),
+				cta: translate( 'Upgrade your site security for only an additional:' ),
+				features: [
+					{
+						slug: PRODUCT_JETPACK_BACKUP_T1_YEARLY,
+						text: translate( 'Real-time cloud backups and one-click restores' ),
+					},
+					{
+						slug: PRODUCT_JETPACK_SCAN,
+						text: translate( 'Malware scanning and one-click fixes' ),
+					},
+					{
+						slug: PRODUCT_JETPACK_ANTI_SPAM,
+						text: translate( 'Block spam in comments and forms' ),
+					},
+				],
+			};
+		}
+
+		return {};
+	}, [ upsellSlug ] );
+
+	const productName = productItem?.displayName;
+	const upsellName = upsellItem?.displayName;
+
+	const productPriceObj = useItemPrice( siteId, productItem );
+	const upsellPriceObj = useItemPrice( siteId, upsellItem );
+	const priceDelta =
+		( upsellPriceObj?.discountedPrice || upsellPriceObj?.originalPrice ) -
+		( productPriceObj?.discountedPrice || productPriceObj?.originalPrice );
+	const isLoadingPrice = productPriceObj?.isFetching || upsellPriceObj?.isFetching;
+
+	useEffect( () => {
+		const dontShowUpsell =
+			! isLoadingUpsellPageExperiment && experimentAssignment?.variationName !== 'treatment';
+
+		if ( ( ! upsellSlug || isUpsellOwned || dontShowUpsell ) && productCheckoutURL ) {
+			page.redirect( productCheckoutURL );
+		}
+	}, [
+		upsellSlug,
+		isUpsellOwned,
+		productCheckoutURL,
+		isLoadingUpsellPageExperiment,
+		experimentAssignment,
+	] );
+
+	if ( ! upsellSlug || isFetchingPurchases || isUpsellOwned ) {
+		return null;
+	}
+
+	return (
+		<>
+			{ siteId && <QuerySiteProducts siteId={ siteId } /> }
+			{ siteId && <QueryIntroOffers siteId={ siteId } /> }
+
+			<Main wideLayout>
+				<h1>
+					{ translate( 'Nice choice, we added %(productName)s to your cart.', {
+						args: {
+							productName,
+						},
+					} ) }{ ' ' }
+					{ intro }
+				</h1>
+				<p>{ translate( 'Best value' ) }</p>
+				<h2>{ upsellName }</h2>
+				{ description && <p>{ description }</p> }
+				{ features?.length && (
+					<ul>
+						{ features.map( ( { slug, text } ) => (
+							<li key={ slug }>
+								{ productNames[ slug ] || slug } -{ ' ' }
+								{ slug === productSlug ? translate( 'Already in your cart' ) : text }
+							</li>
+						) ) }
+					</ul>
+				) }
+				{ ! isLoadingPrice && priceDelta > 0 && (
```

Comment:
Is there a way to show a loading state here? (may be in a separate PR already)
When this loads the CTA buttons move down - I missed clicking on them a few times :)